### ISSUE TITLE
very bad classification accuracy

### ISSUE THREAD

**Issue by Rahul6818** [facebookresearch/fastText#570]
Hi, 
I am working on chatbot and was trying fastText for classification. 
My training data comprises of around 300 classes and 5 training examples per class. Most of the time it return the same class of most of the queries. 
So what are the area of improvements here. Please suggest.

**Comment by sayak1711** [comment]
I have faced similar issue. Accuracy is not good.

**Comment by Rahul6818** [comment]
@sayak1711  how did you solved it then?

**Comment by sayak1711** [comment]
I haven't fully able to solve it however I have realized that predictions results vary on giving different subsets of the same input for prediction. Upon giving smaller subset results seem to differ than from the entire input. Also issue [455](https://github.com/facebookresearch/fastText/issues/455) might be useful

**Comment by adam2326** [comment]
5 training examples per class?  You should have more like thousands of
examples per class if want you accuracy from fasttext.  Even a simpler
model would most likely require more examples before getting anything
accurate.

**Comment by EdouardGrave** [comment]
Hi @Rahul6818 and @sayak1711,

As mentioned by @adam2326, 5 examples per class is very small, and you should not expect to get high accuracy with such a small training set.

However, you can try to improve the performance of the model by doing the following. First, you should try to increase the number of epochs and tune the learning rate, by using the command line options `-epoch` and `-lr`. You can also try to improve the performance of the model by using pre-trained word vectors to initialize the model. For English, you can use the models that are available [here](https://fasttext.cc/docs/en/english-vectors.html), while for other languages, you can use the models that are available [here](https://fasttext.cc/docs/en/crawl-vectors.html). Please use the command line option `-pretrainedVectors` to do so, and change the dimension of the model to 300 by using `-dim 300`.

Best,
Edouard.

**Comment by sayak1711** [comment]
Hi Edouard. I have used about 2970 examples per class. There were 2 classes and they were related to the financial domain. However thanks for your suggestion.

**Comment by Rahul6818** [comment]
Thanks @EdouardGrave . Will try your suggestions.

**Comment by Rahul6818** [comment]
@EdouardGrave  I tried loading the english vector model [wiki-news-300d-1M] but its size is around 2.1GB. So being so heavy whenever i try to load it. My system gets too slow and start hanging. I there any alternative. I am tried loading vector like below - 

import json
MODEL_DIR = "../wiki-news-300d-1M.vec"
vectors = {}
with open(MODEL_DIR,"r",encoding='utf8') as f:
    lines = f.readlines()
    
lines = [line.split() for line in lines]
vocab_size,vector_size = int(lines[0][0]), int(lines[0][1])

for line in lines[1:]:
    vectors[line[0]] = map(lambda x:float(x),line[1:])

**Comment by EdouardGrave** [comment]
Hi @Rahul6818,

You can try to reduce the size of the pre-trained word vectors, by only loading the top-k words. This can be achieved by using
```
> head -n 200000 wiki-news-300d-1M.vec > wiki-news-small.vec
```
and using `wiki-news-small.vec` as pre-trained vectors.

Best,
Edouard
